src/TechCounts.cpp

```diff
@@ -0,0 +1,319 @@
+					}
+				}
+			}
+		}
+
+		// Check for brackets
+		if(currentRow.noteCount >= 2)
+		{
+			if(currentRow.whereTheFeetAre[StepParity::LEFT_HEEL] != -1 && currentRow.whereTheFeetAre[StepParity::LEFT_TOE] != -1)
+			{
+				out[TechCountsCategory_Brackets] += 1;
+			}
+
+			if(currentRow.whereTheFeetAre[StepParity::RIGHT_HEEL] != -1 && currentRow.whereTheFeetAre[StepParity::RIGHT_TOE] != -1)
+			{
+				out[TechCountsCategory_Brackets] += 1;
+			}
+		}
+
+		// Check for up footswitches
+		for(int c : layout.upArrows)
+		{
+			if(isFootswitch(c, currentRow, previousRow, elapsedTime))
+			{
+				out[TechCountsCategory_UpFootswitches] += 1;
+				out[TechCountsCategory_Footswitches] += 1;
+			}
+		}
+		// Check for down footswitches
+		for(int c: layout.downArrows)
+		{
+			if(isFootswitch(c, currentRow, previousRow, elapsedTime))
+			{
+				out[TechCountsCategory_DownFootswitches] += 1;
+				out[TechCountsCategory_Footswitches] += 1;
+			}
+		}
+		
+		// Check for sideswitches
+		for(int c: layout.sideArrows)
+		{
+			if(isFootswitch(c, currentRow, previousRow, elapsedTime))
+			{
+				out[TechCountsCategory_Sideswitches] += 1;
+			}
+		}
+		
+		// Check for crossovers
+		int leftHeel = currentRow.whereTheFeetAre[StepParity::LEFT_HEEL];
+		int leftToe = currentRow.whereTheFeetAre[StepParity::LEFT_TOE];
+		int rightHeel = currentRow.whereTheFeetAre[StepParity::RIGHT_HEEL];
+		int rightToe = currentRow.whereTheFeetAre[StepParity::RIGHT_TOE];
+		
+		int previousLeftHeel = previousRow.whereTheFeetAre[StepParity::LEFT_HEEL];
+		int previousLeftToe = previousRow.whereTheFeetAre[StepParity::LEFT_TOE];
+		int previousRightHeel = previousRow.whereTheFeetAre[StepParity::RIGHT_HEEL];
+		int previousRightToe = previousRow.whereTheFeetAre[StepParity::RIGHT_TOE];
+		
+		// Check for the following:
+		// - We moved the right foot on this row,
+		// - we moved the left foot on this row,
+		// - we didn't move the right foot on this row
+		// - Is the position of the right foot farther left than the left foot
+		// - If so, check the row before the previous row for the following:
+		//   - Was the right foot on a difference position
+		//   - Was the right foot farther right than the left?
+		//     - If so, then this was a full crossover (like RDL, starting on right foot)
+		//     - otherwise, then this was probably a half crossover (like UDL, starting on right foot)
+		if(rightHeel != -1 && previousLeftHeel != -1 && previousRightHeel == -1)
+		{
+			StepParity::StagePoint leftPos = layout.averagePoint(previousLeftHeel, previousLeftToe);
+			StepParity::StagePoint rightPos = layout.averagePoint(rightHeel, rightToe);
+			
+			if(rightPos.x < leftPos.x)
+			{
+				if(i > 1)
+				{
+					const StepParity::Row & previousPreviousRow = rows[i - 2];
+					if(previousPreviousRow.whereTheFeetAre[StepParity::RIGHT_HEEL] != rightHeel)
+					{
+						StepParity::StagePoint previousPreviousRightPos = layout.columns[previousPreviousRow.whereTheFeetAre[StepParity::RIGHT_HEEL]];
```

Comment:
I'm getting a crash here on the "BREACH PROTOCAL" simfile I attached in the other thread because `previousPreviousRow.whereTheFeetAre[StepParity::RIGHT_HEEL]` is `-1`.